app/javascript/src/Common/components/SelectWithModal.tsx

```diff
@@ -102,22 +99,19 @@ const SelectWithModal = <T extends IRecord>({
   }, [page, shouldHaveModal, modalOpen])
 
   useEffect(() => {
-    if (!fetchItems) {
+    if (!fetchItems || !modalOpen) {
```

Comment:
This is to prevent fetching if the modal is being closed (e.g. when the query is cleared upon the modal closing).